projects/gloo/pkg/plugins/adaptive_concurrency/plugin.go

```diff
@@ -0,0 +1,221 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// Variable declarations with defaults
+	var (
+		sampleAggregatePercentile      float64
+		concurrencyLimitExceededStatus uint32
+	)
+
+	// Set sample_aggregate_percentile with default
+	if adaptiveConcurrency.GetSampleAggregatePercentile() != nil {
+		sampleAggregatePercentile = adaptiveConcurrency.GetSampleAggregatePercentile().GetValue()
+	} else {
+		sampleAggregatePercentile = DefaultSampleAggregatePercentile
+	}
+
+	// Set concurrency limit exceeded status with default of 503 for non-error codes
+	status := adaptiveConcurrency.GetConcurrencyLimitExceededStatus()
+	if status == 0 || status < 400 {
+		concurrencyLimitExceededStatus = DefaultConcurrencyLimitExceededStatus
+	} else {
+		concurrencyLimitExceededStatus = status
+	}
+
+	gradientControllerConfig := &envoy_adaptive_concurrency_v3.GradientControllerConfig{
+		ConcurrencyLimitParams:    concurrencyLimitParams,
+		MinRttCalcParams:          minRttCalcParams,
+		SampleAggregatePercentile: &typev3.Percent{Value: sampleAggregatePercentile},
+	}
+
+	out := &envoy_adaptive_concurrency_v3.AdaptiveConcurrency{
+		ConcurrencyControllerConfig: &envoy_adaptive_concurrency_v3.AdaptiveConcurrency_GradientControllerConfig{
+			GradientControllerConfig: gradientControllerConfig,
+		},
+		ConcurrencyLimitExceededStatus: &typev3.HttpStatus{
+			Code: typev3.StatusCode(concurrencyLimitExceededStatus),
+		},
+	}
+
+	return out, nil
+}
+
+func translateConcurrencyLimitParams(in *adaptive_concurrency.FilterConfig) (*envoy_adaptive_concurrency_v3.GradientControllerConfig_ConcurrencyLimitCalculationParams, error) {
+	if in.GetConcurrencyUpdateIntervalMillis() == 0 {
+		return nil, ErrConcurrencyUpdateIntervalMillisMissing
+	}
+
+	// Variable declarations with defaults
+	var maxConcurrencyLimit uint32
+
+	// Set max_concurrency_limit with default
+	if in.GetMaxConcurrencyLimit() != nil {
+		maxConcurrencyLimit = in.GetMaxConcurrencyLimit().GetValue()
+	} else {
+		maxConcurrencyLimit = DefaultMaxConcurrencyLimit
+	}
+
+	out := &envoy_adaptive_concurrency_v3.GradientControllerConfig_ConcurrencyLimitCalculationParams{
+		ConcurrencyUpdateInterval: millisToDuration(in.GetConcurrencyUpdateIntervalMillis()),
+		MaxConcurrencyLimit:       &wrapperspb.UInt32Value{Value: maxConcurrencyLimit},
+	}
+
+	return out, nil
+}
+
+func translateMinRttCalcParams(in *adaptive_concurrency.FilterConfig_MinRoundtripTimeCalculationParams) (*envoy_adaptive_concurrency_v3.GradientControllerConfig_MinimumRTTCalculationParams, error) {
+	if in == nil {
+		return nil, ErrMinRttCalcParamsMissing
+	}
+
+	intervalMillis := in.GetIntervalMillis()
+	fixedValueMillis := in.GetFixedValueMillis()
+
+	// If both are set, Interval is used
+	if intervalMillis == 0 && fixedValueMillis == 0 {
+		return nil, ErrIntervalMissing
+	}
+
+	// Variable declarations with defaults
+	var (
+		requestCount     uint32
+		minConcurrency   uint32
+		jitterPercentile float64
+		bufferPercentile float64
```

Comment:
nit: Initialize these to the default to make the subsequent if conditions simpler